packages/system/src/Ord/operations.ts

```diff
@@ -94,3 +95,25 @@ export function inverted<A>(O: Ord<A>) {
 export function getEqual<A>(O: Ord<A>): Equal<A> {
   return makeEqual((x, y) => O.compare(x, y) === 0)
 }
+
+/**
+ * Given a tuple of `Ord`s returns an `Ord` for the tuple
+ */
+export function tuple<T extends ReadonlyArray<Ord<any>>>(
```

Comment:
```ts
export function tuple<T extends ReadonlyArray<Ord<any>>>(
  ...ords: T
): Ord<
  ForcedTuple<{
    [K in keyof T]: T[K] extends Ord<infer A> ? A : never
  }>
> {
  return makeOrd((x, y) => {
    return pipe(
      ords,
      A.findFirstMapWithIndex((i, ord) => {
        const r = ord.compare(x.get(i), y.get(i))
        return r !== 0 ? O.some(r) : O.none
      }),
      O.getOrElse<Ordering>(() => 0)
    )
  })
}
```

Don't think it's really better though?

Reply:
Kind of ok both ways, the current one seems slightly better perf wise